src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -222,6 +233,19 @@ private SafeCurlMultiHandle CreateAndConfigureMultiHandle()
                         Interop.Http.CURLMoption.CURLMOPT_PIPELINING,
                         (long)Interop.Http.CurlPipe.CURLPIPE_MULTIPLEX));
                 }
+
+                // Configure max connections per host if it was changed from the default
+                if (_associatedHandler.MaxConnectionsPerServer < int.MaxValue) // int.MaxValue considered infinite, mapping to libcurl default of 0
```

Comment:
Can MaxConnectionsPerServer be changed after this executes?  If so, does this run again?